src/shared/bus-util.c

```diff
@@ -709,3 +711,35 @@ int bus_property_get_string_set(
 
         return bus_message_append_string_set(reply, *s);
 }
+
+void bus_log_caller(sd_bus_message *message, const char *method) {
+        _cleanup_(sd_bus_creds_unrefp) sd_bus_creds *creds = NULL;
+        _cleanup_free_ char *unit_pid = NULL;
+        const char *comm = NULL, *caller = NULL, *unit_creds = NULL;
+        pid_t pid;
+
+        assert(message);
+        assert(method);
+
+        if (sd_bus_query_sender_creds(message, SD_BUS_CREDS_PID|SD_BUS_CREDS_AUGMENT|SD_BUS_CREDS_COMM|SD_BUS_CREDS_UNIT, &creds) < 0)
+                return;
+
+        /* We need at least the PID, otherwise there's nothing to log, the rest is optional */
+        if (sd_bus_creds_get_pid(creds, &pid) < 0)
+                return;
+
+        (void) sd_bus_creds_get_comm(creds, &comm);
+        (void) sd_bus_creds_get_unit(creds, &unit_creds);
+
+        caller = unit_creds;
+
+        if (!unit_creds) {
+                cg_pid_get_unit(pid, &unit_pid);
```

Comment:
`SD_BUS_CREDS_AUGMENT|SD_BUS_CREDS_UNIT` (which you specify above) does exactly this already. Please drop this redundant step hence.

Reply:
It added that because unit extraction from credentials after `daemon-reload` and `daemon-reexec`  actions failed though I don't understand why. 

I will check again after rebasing this branch on top of main.